docs/specs/idp-account-linking.md

```diff
@@ -0,0 +1,528 @@
+        type: google
+        link_by:
+          identity: "email"
+          idp_claim: "/email"
+```
+
+Assume now there is an existing authgear user with the following identities and authenticators:
+
+- User A
+  - Email Identity: a@example.com
+  - Primary Password Authenticator
+
+And now, the user tries to sign up with a new google account, which has an email `a@example.com` in the google user info. And authgear matched that oauth identity to the existing login ID `a@example.com`.
+
+1. The user should first select a conflicted identity, in this example, there is only one conflicted identity `Email: a@example.com`. So this identity should be selected by the user.
+2. After `Email: a@example.com` being selected, the user will switch to a login flow. Which was specified by `on_conflict.login_flow` And `on_conflict.target_step`:
+
+```yaml
+on_conflict:
+  action: login_and_link
+  login_flow: default
+  target_step: step_1
+```
+
+3. The login flow `default` will be executed, and starting from the specified step `step_1`.
+4. The selected identity `Email: a@example.com` will be automatically used to pass the first identity step.
+
+```yaml
+- identification: email
+  steps:
+    - name: authenticate_primary_email
+      type: authenticate
+      one_of:
+        - authentication: primary_password
+```
+
+5. Then, the login flow will continue. The user has to enter primary password to pass the authentication.
+6. After user entered password, the login flow was completed. Now, the original signup flow will be continued.
+
+```yaml
+- identification: email
+  steps:
+    - name: email_setup_primary_email
+      type: create_authenticator
+      one_of:
+        - authentication: primary_password
+- identification: oauth # <-- Resume here
+  on_conflict:
+    action: login_and_link
+    login_flow: default
+    target_step: step_1
+  steps:
+    - name: oauth_setup_primary_email
+      type: create_authenticator
+      one_of:
+        - authentication: primary_password
+          steps:
+            - type: create_authenticator
+              one_of:
+                - authentication: secondary_totp
+                  steps:
+                    - type: view_recovery_code
+```
+
+7. As the next step of the original flow is create primary password authenticator, the user will need to created primary authenticator.
+8. As the original user already has a primary password authenticator, the step will be skipped.
+9. And the next step will be create `secondary_totp`. As the user don't have a secondary totp, the user should create a totp in this step.
+10. Finally, all created identities and authenticators will be added to the existing user, together with the new oauth identity.
+
+Resulting user:
+
+- User A
+  - Email Identity: a@example.com
+  - OAuth Identity: Google (email:a@example.com)
+  - Primary Password Authenticator
+  - Secondary TOTP Authenticator
+
+## Q&A
+
+- Why we need to login the user before linking the account?
```

Comment:
Use ### for each question. And use a paragraph for each answer.

```
### Why ...

Answer in a paragraph.
```

Remember to re-generate the TOC.